### PR TITLE
PR: Correclty update visualisation after internal change.

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description
At the moment visualizations don't get a data update after we used the cycling shortcut or the visualization chooser until they go through a show/hide cycle. This PR fixes that by adding enable/disable events whenever a visualization on a node changes. 

### Important Notes
Based on #786, might also solve #865, #864

Refactoring of `visualisation::container::FRP` to use the new macro generated FRP structs.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has been tested where possible.
- [x] All code has been profiled where possible.

### CODE REVIEW

**Comment by MichaelMauderer** [comment]
@farmaazon @wdanilo  Could you check whether this change makes sense from the controller side? Is it sane to emit a pair of `visualization_disabled` / `visualization_enabled` events to trigger the controller updating the data for a visualization on a node? Or should we design a better API that informs the controller about graph-editor internal changes to a node's visualization.

File: src/rust/ide/view/graph-editor/src/lib.rs
Comment (farmaazon) [inline src/rust/ide/view/graph-editor/src/lib.rs:991]: I'd rather opt for making "visualization_changed" a part of the GraphEditor API, and do tricks on the integration level. But this can take a bit of time, so for now please make an issue for refactoring this, and put here TODO with issue number.

**Review by farmaazon — CHANGES_REQUESTED** [review]

File: src/rust/ide/view/graph-editor/src/component/visualization/container.rs
Comment (MichaelMauderer) [reply src/rust/ide/view/graph-editor/src/component/visualization/container.rs]: I've refactored this to be a bit nicer and void the memory leak. Unfortunately, filter doesn't work here, as the `definition` needs to be wrapped in an `Option` because it does not itself implement `Default.`

File: src/rust/ide/view/graph-editor/src/lib.rs
Comment (MichaelMauderer) [reply src/rust/ide/view/graph-editor/src/lib.rs]: Added the task #896  and the todo in the doc.

**Review by wdanilo — APPROVED** [review]

**Comment by MichaelMauderer** [comment]
While not blocked, this is waiting for confirmation form @farmaazon before being merged.

**Review by farmaazon — APPROVED** [review]

**Pull request merged**
